packages/11ty/_includes/web-components/image-sequence/index.js

```diff
@@ -311,55 +356,81 @@ class ImageSequence extends LitElement {
    * 
    **/ 
   willUpdate(changedProperties) {
+    // Determine the animation indices, preload them, and then do the rotation 
     if (changedProperties.has('rotateToIndex') && this.rotateToIndex!==false) {
-      this.performRotation(this.rotateToIndex)
+      const frameCount = this.rotateToIndex - this.index
+      const animationIndices = Array(frameCount).fill(0).map( (_,i) => this.index + i + 1 )
+      this.#preloadImages( animationIndices ).then( this.performRotation(this.rotateToIndex) )
     }
 
+    // Draws `animationIndex` directly to canvas (for use )
+    if (changedProperties.has('animationIndex')) {
+      if (!this.animationIndex) { return }
+      if (this.images[this.animationIndex] === null) { return }
+
+      this.#paintCanvas(this.images[this.animationIndex])
+    } 
+
     if (changedProperties.has('index') && this.someImagesLoaded) {
-      this.#preloadImages()
 
       if (this.bufferReady) {
-        this.#paintCanvas()      
+        this.#paintCanvas()
+        return      
       }
+
+      this.#preloadImages().then( () => {
+        this.performRotation(this.index)
+      })
+
     }
 
+    // Load enough to prepare for interaction
     if (changedProperties.has('visible') && !this.visible) {
-      this.#preloadImages()
+      this.#preloadImages().then( () => this.#paintCanvas() )
     }
   }
 
   /**
    * @function #preloadImages
    * 
-   * Loads the k images behind and ahead of this.index
+   * Loads images of this.indexesToBuffer -- ahead and behind `this.index`
+   * 
+   * @returns {Promise} - Promise resolution for all preloads
+   * 
    **/
-  #preloadImages() {
+  #preloadImages(indices) {
     if (!this.images.some(i => i === null)) { return }
 
-    // Really just making buffer counting ergonomic / readable here
-    // const indexesToBuf = Array(this.bufferSize).fill(0).map( (_,i) => ( this.images.length + this.index + i - Math.round(this.bufferSize/2) ) % this.images.length )
-    this.images.forEach( (image,i) => {
+    const indexesToLoad = indices ?? this.indexesToBuffer
+    const fetches = this.images.map( (image,i) => {
       // Skip anything out of our range or already loaded
-      if ( !this.indexesToBuffer.includes(i) || image !== null ) {
-        return
+      if ( !indexesToLoad.includes(i) || image !== null ) {
+        return null
       } 
 
       const url = this.imageUrls[i]
-      this.#fetchImage(url,i)
-    })
+      return this.#fetchImage(url,i)
+    }).filter( ftch => ftch )
```

Comment:
```suggestion
    }).filter((image) => image)
```

Reply:
See above, refined these to reflect your review